autorest/codegen/__init__.py

```diff
@@ -142,15 +143,16 @@ def _get_credential_param(self, azure_arm, credential, credential_default_policy
         credential_scopes = self._get_credential_scopes(credential)
         credential_key_header_name = self._autorestapi.get_value('credential-key-header-name')
 
-        if credential_default_policy_type == "BearerTokenCredentialPolicy":
+        if credential_default_policy_type in ["ARMChallengeAuthenticationPolicy", "BearerTokenCredentialPolicy"]:
```

Comment:
Preferably, we only have `"ARMChallengeAuthenticationPolicy"` and `"BearerTokenCredentialPolicy"` in one place.

Implementing the property I talked about earlier will get you this behavior

Reply:
```suggestion
        if (azure_arm, credential_default_policy_type ) in [(True, "ARMChallengeAuthenticationPolicy"), (False, "BearerTokenCredentialPolicy")]:
```
Do you mean to distinguish data-plan and mgmt-plan like up?